### PR TITLE
Model validations PR

### PR DESCRIPTION
**Pull request by AmaduKamara**
# Blog App: validations, model specs, and n+1 problems
🧑‍💻🧑‍💻🧑‍💻
Added the following validations:

 - For the User model:
    - Name is not blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all my controllers.
- Made use of the N+1 problem solution when fetching all posts and their comments for a user.

### Hi Reviewer, @Reviewer!
There is a special issue on my linter configuration I don't understand. It's actually working fine locally, robocop working fine but Linters not reflecting on GitHub. I've tried several solutions, readded the linter files for ror and it's the same from the very start of this repo and linter configuration. Please clone the project to test it.

I sent this same message to the previous reviewer. Thanks for your time👍

### CODE REVIEW

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @AmaduKamara  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.



### ♻️ Request changes ♻️

- [x] _Check the comments under the review._
image

- [x] For solve the n+1 problem you should write `posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts.order(created_at: :desc)` . You can still add created_at desc end of the code.
___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.order(created_at: :desc)
+    # @posts = @user.posts.order(created_at: :desc)
```
Comment (MrBrN197) [inline app/controllers/posts_controller.rb:4]: - [x] [OPTIONAL] You can remove this previous code as it is unnecessary now.

**Review by MrBrN197 — APPROVED** [review]
## Hi, @AmaduKamara  👋

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (AmaduKamara) [reply app/controllers/posts_controller.rb:4]: Thanks, bro. I removed it from the branch.

**Pull request merged**